keeper/query_server_portal.go

```diff
@@ -67,6 +70,36 @@ func (k portalQueryServer) Peers(ctx context.Context, req *portal.QueryPeers) (*
 	return &portal.QueryPeersResponse{Peers: peers}, err
 }
 
+func (k portalQueryServer) SupportedBridgingPaths(ctx context.Context, req *portal.QuerySupportedBridgingPaths) (*portal.QuerySupportedBridgingPathsResponse, error) {
+	if req == nil {
+		return nil, types.ErrInvalidRequest
+	}
+
+	// NOTE: We have to typecast here because protoc-gen-grpc-gateway doesn't
+	// support it via gogoproto.
+	destinationChainId := uint16(req.DestinationChainId)
```

Comment:
It looks like `req.DestinationChainId` is a `uint32` but then we typecast to a `uint16`, why do we need to do this? Can't we just use `uint16` in the type definition? 

I'm guessing the comment here is somehow related but it isn't very insightful.

Reply:
Okay so I believe this answers a bunch of questions here, but Protobuf only supports `uint32` and `uint64` natively. We can utilize `gogoproto` to typecast to `uint16` when (un)marshalling from the store. However, this doesn't work when generating the gRPC gateway types which are used by the Cosmos SDK to wire up the handlers. Therefore, we have to do the typecast here ourselves 😄